### PR TITLE
Introduce a `Handler` alias for `ExceptT ServantErr IO`

### LINKED ISSUE #434: Introduce a type alias `Handler`
``` haskell
type Handler = ExceptT ServantErr IO
```

This would make it shorter to write type signatures. It would also make it less painful to replace this with a custom type later on.

### PR DESCRIPTION
**Pull request by lcycon**
Fixes #434

### CODE REVIEW

**Comment by lcycon** [comment]
Ack. This is gonna fail methinks. Ambiguous `Handler`. Not sure why this doesn't happen on my machine..

**Comment by 3noch** [comment]
Maybe `HandlerM` to match `ClientM`?

**Comment by lcycon** [comment]
That feels more natural to me, but I was just going off the issue referenced above.

**Comment by kosmikus** [comment]
Sorry for bikeshedding: But if it really is called `ClientM` now (I'm not particularly fond of that name), then I think the corresponding concept for the server should also be called `ServerM`.

**Comment by lcycon** [comment]
You know, the more I look at some of my apps (modified to build against the
head of my branch), the more I like the look of `Handler`.

That said, all but one of my servant using applications use a custom Monad
stack, so I can't say it's a huge difference..

**Comment by alpmestan** [comment]
I like the simplicity of `Handler` but a conflict with an existing type in `Control.Exception` is really problematic. I'm really not fond of `ServerM` either. This is probably going to be confusing, when put next to `Server`/`ServerT`.

**Comment by soenkehahn** [comment]
My vote is (still) for `Handler`.

- I think it's the most fitting.
- I'm also not *that* happy with `ClientM`. I think it shouldn't affect our choice in `servant-server`. We may even want to rename `ClientM`.
- `snap` uses the same name (`Handler`) for something similarish.
- I don't see a big problem with the name collision with `Control.Exception`. (Maybe this warrants a major version bump, since `Control.Exception` is very widely used. But I'm fine with that.)

File: servant-server/src/Servant/Server/Experimental/Auth.hs [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+{-# LANGUAGE DeriveDataTypeable    #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE UndecidableInstances  #-}
```
Comment (soenkehahn) [inline servant-server/src/Servant/Server/Experimental/Auth.hs]: This seems like two formatting tools having an edit-war. :) @lcycon, can you remove this, please?

File: servant-server/src/Servant/Server/Internal/Handler.hs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module Servant.Server.Internal.Handler where
+
+import           Control.Monad.Trans.Except         (ExceptT)
+import           Servant.Server.Internal.ServantErr (ServantErr)
+
+type Handler = ExceptT ServantErr IO
```
Comment (soenkehahn) [inline servant-server/src/Servant/Server/Internal/Handler.hs]: I would consider putting `Handler` into `Servant.Server.Internal.ServantErr`.

**Comment by soenkehahn** [comment]
Apart from my comments, LGTM.

**Comment by kosmikus** [comment]
I can live very well with `Handler` if that's what everyone wants, but I'd like to consider renaming `ClientM` then. There is some value to consistent naming ...

File: servant-server/src/Servant/Server/Experimental/Auth.hs (hunk 1 above)
Comment (lcycon) [inline servant-server/src/Servant/Server/Experimental/Auth.hs]: Ack. I thought I turned that off. Emacs has a mind of its own ;)

File: servant-server/src/Servant/Server/Internal/Handler.hs (hunk 2 above)
Comment (lcycon) [inline servant-server/src/Servant/Server/Internal/Handler.hs]: I had the same thought, but couldn't decide. I'll move it over.

**Comment by mbg** [comment]
I agree that it makes sense to rename `ClientM` as well to maintain consistency. I didn't think about the naming too much when I introduced it since I was only looking for an alias for the monad stack we had initially in #432.

**Comment by soenkehahn** [comment]
@lcycon: Could you squash your commits and rebase on master, please? After that, I'll merge.

Squashing, since there's a failing commit in between and some whitespace back-and-forth. I'd like to keep that out of the git history, if possible.

**Comment by lcycon** [comment]
@soenkehahn 10-4, gimme a moment

**Comment by lcycon** [comment]
@soenkehahn There ya go

**Comment by soenkehahn** [comment]
@lcycon: I've added something to the changelog: d4c6f67.

Thanks a lot!

**Comment by 3noch** [comment]
Did ya'll end up deciding that `ClientM` needs to change too?

**Comment by soenkehahn** [comment]
@3noch: No. I think this should happen in a separate issue or PR.

**Comment by 3noch** [comment]
@soenkehahn I was just curious if we need an issue to track that work, if it was desired.

**Comment by soenkehahn** [comment]
@3noch: Feel free to open an issue.

**Pull request merged**